### PR TITLE
feat: volunteer validator

### PR DESCRIPTION
**Pull request by JoowonYun**
## Effects

- [ ] Soft update
- [x] Breaking change
- [ ] Chain fork related

## Major Reviewer
@yoosah 

## Background

- Existence of validators who want to participate without reward.

## Summary

- Zero reward validators cannot receive delegation.
- The zero reward validator's commission goes to the community pool
- Zero reward validators are registered through proposals.
- Zero reward validator participates in block propose regardless of the active set

## Checklist

- [ ] Backward compatible?
- [ ] Test enough in your local environment?
- [ ] Add related test cases?

### CODE REVIEW

**Comment by psy2848048** [comment]
Max validator does not change by zero reward validator's in & out

1. Parameter changed, `MaxValidators == 5`: https://github.com/xpladev/xpla/blob/test/special-validator/integration_test/integration_test.go#L1218-L1222
2. Added zero reward validator -> expected to be `MaxValidators == 5 + 1`
But `MaxValidators` is still 5. If a zero reward validator joins, another general validator would be kicked out.

Please check the test code [here](https://github.com/xpladev/xpla/blob/test/special-validator/integration_test/integration_test.go#L1213)

**Comment by JoowonYun** [comment]
> Max validator does not change by zero reward validator's in & out
> 
> 1. [Parameter changed](https://github.com/xpladev/xpla/blob/test/special-validator/integration_test/integration_test.go#L1218-L1222) `MaxValidators == 5`
> 2. Added zero reward validator -> expected to be `MaxValidators == 5 + 1`
>    But `MaxValidators` is still 5. If a zero reward validator joins, another general validator would be kicked out.
> 
> Please check the test code [here](https://github.com/xpladev/xpla/blob/test/special-validator/integration_test/integration_test.go#L1213)

'MaxValidators' is a params. So it can't be change.  You can check all validators which is status bond or prevote count which is (MaxValidators +1)

**Comment by psy2848048** [comment]
Consensus failure issue

- Initial state: 6 validators + 2 general nodes
- Set `MaxValidators` to 7
- Added 1 validator as a zero reward validator
- Try to add 1 general validator by `NewMsgCreateValidator`
- Raised Consensus failure: `CONSENSUS FAILURE!!! err="validator EndBlock updates already set by a previous module" module=consensus`

**Comment by psy2848048** [comment]
New Consensus failure issue

- Test case: applying simple unregister zero reward validator (`Test12_GeneralZeroRewardValidatorRegistryUnregistryDelegation`)
- Msg: `0axpla is smaller than 1000000000000000000axpla`

File: x/zeroreward/keeper/val_state_change.go [hunk 1]
```diff
@@ -87,25 +81,37 @@ func (k Keeper) ZeroRewardValidatorUpdates(ctx sdk.Context) (updates []abci.Vali
...
+			if prevPower != newPower {
+				updates = append(updates, abci.ValidatorUpdate{
+					PubKey: tmProtoPk,
+					Power:  newPower,
+				})
+			}
 
 			zeroRewardValidator.Power = newPower
-			k.SetZeroRewardValidator(ctx, addr, zeroRewardValidator)
+			k.SetValidator(ctx, addr, zeroRewardValidator)
```
Comment (psy2848048) [inline x/zeroreward/keeper/val_state_change.go:114]: Here?

**Review by psy2848048 — APPROVED** [review]

**Review by yoosah — COMMENTED** [review]
@psy2848048

File: integration_test/util_test.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
+package integrationtest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
```
[the remark this answers is not in the document]
Comment (psy2848048) [reply integration_test/util_test.go]: https://github.com/xpladev/xpla/pull/81/commits/859f8d17439e87c5982e4ae4fc6183b8336d8a96

**Review by caramis — APPROVED** [review]
LGTM

**Pull request merged**